packages/venia-ui/lib/RootComponents/Search/categoryFilters.js

```diff
@@ -0,0 +1,68 @@
+import React, { useCallback } from 'react';
+import { X as CloseIcon } from 'react-feather';
+import gql from 'graphql-tag';
+import { Query } from '@magento/venia-drivers';
+
+import { mergeClasses } from '../../classify';
+import Icon from '../../components/Icon';
+import getQueryParameterValue from '../../util/getQueryParameterValue';
+import defaultClasses from './categoryFilters.css';
+
+const GET_CATEGORY_NAME = gql`
+    query getCategoryName($id: Int!) {
+        category(id: $id) {
+            name
+        }
+    }
+`;
+
+const CategoryFilters = props => {
+    const { categoryId, executeSearch, history, location } = props;
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    const renderResult = useCallback(resultProps => {
+        const { data, error, loading } = resultProps;
+
+        if (error) return null;
+        if (loading) return 'Loading...';
+        return data.category.name;
+    }, []);
+
+    const handleClearCategoryFilter = useCallback(() => {
+        const inputText = getQueryParameterValue({
+            location,
+            queryParameter: 'query'
+        });
+
+        if (inputText) {
+            executeSearch(inputText, history);
+        }
+    }, [executeSearch, history, location]);
+
+    return (
+        <div className={classes.root}>
+            <button
+                className={classes.filter}
+                onClick={handleClearCategoryFilter}
+            >
+                <small className={classes.text}>
+                    <Query
+                        query={GET_CATEGORY_NAME}
+                        variables={{ id: categoryId }}
+                    >
+                        {renderResult}
```

Comment:
Would it be more appropriate to put all the html as the query child instead of just the name? In the event of an error from GQL we don't want to render a blank div anyways.

Reply:
Rather than a blank container, it's actually a button with functionality (and a close icon). If the query were to fail, we'd still want the button and the close icon to be there.

I think we should preserve the original work here.